crypto/objects/o_names.c

```diff
@@ -83,7 +91,7 @@ int OBJ_NAME_new_index(unsigned long (*hash_func) (const char *),
             return (0);
         }
         name_funcs->hash_func = lh_strhash;
-        name_funcs->cmp_func = OPENSSL_strcmp;
+        name_funcs->cmp_func = obj_strcmp;
```

Comment:
I suppose this is okay, but I'd kind prefer to see the else/define at 28/29 removed and an ifdef here.